src/sonic-config-engine/portconfig.py

```diff
@@ -430,7 +430,8 @@ def parse_platform_json_file(hwsku_json_file, platform_json_file):
             if child_port in hwsku_entry:
                 for key, item in hwsku_entry[child_port].items():
                     if key in OPTIONAL_HWSKU_ATTRIBUTES:
-                        child_ports.get(child_port)[key] = item
+                        for child in child_ports:
```

Comment:
@bobbymcgonigle please make sure platform like which have subport > 1 specified in hwsku.json are not impacted https://github.com/sonic-net/sonic-buildimage/blob/master/device/mellanox/x86_64-mlnx_msn4700-r0/Mellanox-SN4700-O8V48/hwsku.json#L10